aptos-move/aptos-aggregator/src/types.rs

```diff
@@ -37,14 +38,131 @@ pub trait TryFromMoveValue: Sized {
     ) -> Result<Self, Self::Error>;
 }
 
-pub type AggregatorResult<T> = Result<T, AggregatorError>;
+// represents something that should never happen - i.e. a code invariant error,
+// which we would generally just panic, but since we are inside of the VM,
+// we cannot do that.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PanicError(String);
 
-// TODO: Use this instead of PartialVM errors.
-#[derive(Debug)]
-pub enum AggregatorError {
-    WrongVersionID,
+impl ToString for PanicError {
+    fn to_string(&self) -> String {
+        self.0.clone()
+    }
+}
+
+// Wrapping another error, to add a variant that represents
+// something that should never happen - i.e. a code invariant error,
+// which we would generally just panic, but since we are inside of the VM,
+// we cannot do that.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PanicOr<T: std::fmt::Debug> {
+    CodeInvariantError(String),
+    Or(T),
+}
+
+pub type PanicOrResult<T, E> = Result<T, PanicOr<E>>;
```

Comment:
PanicOrResult reads in the wrong way for me😅 how about having MaybePanic, the MaybePanicResult?

Reply:
Then it's even more confusing haha - result or panic or result or orpanic/panicor

Reply:
If you think logically it can be either V, E, or Panic. Pretty good name in my opinion :)

Reply:
I give up, I confused Satya too :) removing this alias, it is not shorter anyways :D